torch/csrc/jit/ir/ir.h

```diff
@@ -1512,5 +1518,68 @@ struct OperatorSet {
   std::unordered_map<Symbol, std::vector<std::shared_ptr<Operator>>> ops;
 };
 
+template <typename T>
+struct OperatorMap {
+  OperatorMap() {}
+  OperatorMap(std::initializer_list<std::pair<std::string, T>> init) {
+    insert(init);
+  }
+
+  bool contains(const Symbol& key) const {
+    return map.count(key) != 0;
+  }
+
+  bool contains(const std::string& key) const {
+    return map.count(Symbol::fromQualString(key)) != 0;
+  }
+
+  void insert(const Symbol& key, T value) {
+    map[key] = value;
+  }
+
+  void insert(const std::string& key, T value) {
+    map[Symbol::fromQualString(key)] = value;
+  }
+
+  void insert(std::initializer_list<std::pair<Symbol, T>> v) {
+    for (auto el : v) {
+      map[el.first] = el.second;
+    }
+  }
+
+  void insert(std::initializer_list<std::pair<std::string, T>> v) {
+    for (auto el : v) {
+      map[Symbol::fromQualString(el.first)] = el.second;
+    }
+  }
+
+  void erase(const Symbol& key) {
```

Comment:
I think the public APIs should only take in `Operator` as keys, and not Symbols/strings (related to the other comment).